dask/dataframe/shuffle.py

```diff
@@ -370,7 +338,12 @@ def set_partition(
             column_dtype=df.columns.dtype,
         )
 
-    df4.divisions = tuple(methods.tolist(divisions))
+    # None and pd.NA values are not sortable
+    divisions = methods.tolist(divisions)
+    if pd.isna(divisions).any():
+        divisions = [np.nan] * len(divisions)
+
+    df4.divisions = tuple(divisions)
```

Comment:
This is, I think, a bit more straightforward and also keeps non-na entries around in `divisions` (though I don't know that we have a test case for this -- it still seems like a reasonable thing to do) 

```suggestion
    divisions = methods.tolist(divisions)
    # None and pd.NA values are not sortable
    df4.divisions = tuple(i if not pd.isna(i) else np.nan for i in divisions)
```